src/arrayMethodSort.js

```diff
@@ -5,7 +5,39 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let compareFunc;
+    let temp;
+    let counter = 0;
+
+    if (!compareFunction) {
+      compareFunc = (a, b) => {
+        if (String(a) < String(b)) {
+          return -1;
+        } else if (String(a) > String(b)) {
+          return 1;
+        }
+
+        return 0;
+      };
```

Comment:
BTW better to move this function to the top level and use it as a default value for `compareFunction`;
```
const defaultCompareFunction = ...
[].__proto__.sort2 = function(compareFunction = defaultCompareFunction) {